### DISCUSSION TITLE
Make typescript plugin official

### DISCUSSION THREAD

**Discussion by zabil** [getgauge/gauge#2551] [Ideas]
cc @sriv, @chadlwilson, @haroon-sheikh 

Hi everyone,

While working on Taiko functional tests, I noticed that @BugDiver's great
TypeScript plugin, [gauge-ts](https://github.com/bugdiver/gauge-ts), is
outdated and still using TypeScript 3. I forked the repo, made a lot of
updates, and got it working with the latest Node versions and TypeScript 5. You
can see the changes here:
https://github.com/BugDiver/gauge-ts/compare/main...zabil:gauge-ts:main.

There are quite a few changes. I think this plugin is a good candidate to
become an official Gauge plugin. I have some time to maintain it. Do you think
it's okay to move it to the getgauge repo and publish it as
`@getgauge/gauge-ts`?

**Comment by bugdiver** [comment]
@zabil  I can transfer the repository under `gauge` organization. I'm Not sure if the npm package can be transferred. Once you have published it in `@getgauge` scode, I can takedown `gauge-ts` from my account.

**Reply by chadlwilson** [reply]
If @BugDiver wants to transfer the repo, that's probably cleanest, yeah. Since he is still an org owner for `getgauge`, he probably has all the permissions to transfer the repo into the org he needs.

We could merge @zabil's changes and sort out release automation afterwards?

**Reply by bugdiver** [reply]
@chadlwilson @zabil  I have already transferred the repository. Github take cares of redirection so creating new projects should work as it, but we can update the release links in the `gauge-repository`.
The docs are also live now on  http://getgauge.github.io/gauge-ts/, This can also be integrated with the official documentation.

**Reply by zabil** [reply]
Ok great. I'll do a bit more testing and send a pull request to @BugDiver's repo and then we can move it to getgauge. Will dig and bit about the npm repo transfer.

**Reply by bugdiver** [reply]
@zabil I have send an invitation to the npm user `getgauge` as a maintainer. Altough I'm not sure which email is linked to this account

**Reply by chadlwilson** [reply]
I know which email is attached, but I don't see an invite. Hmm, curious :-) We need to do some work to tighten up the NPM account and switch to automation tokens for publishing I believe. Wonder if that is related....

**Reply by zabil** [reply]
Yeah I checked too. Don't see the invitation in the associated mail id

**Reply by chadlwilson** [reply]
The NPM settings are now tightened, in case it makes any difference. (Good to do anyway)

**Reply by bugdiver** [reply]
I can see the invitaiton on the package settings page, maybe login into npm and check if there is any notification? 
<img width="819" alt="Screenshot 2024-06-06 at 8 24 07 PM" src="https://github.com/getgauge/gauge/assets/15309877/223441c0-a323-43d2-a671-b9f25bff9417">

**Reply by chadlwilson** [reply]
Already did that -  nothing there either. Maybe remove it and try again.

**Reply by chadlwilson** [reply]
Not sure if you did something else, but got the invite now :-)

**Reply by bugdiver** [reply]
Yeah, I removed and send it again

**Reply by chadlwilson** [reply]
Thanks @BugDiver - it might have been because the npm account had been marked as inactive until I went in there and prodded it a bit :-)

**Reply by chadlwilson** [reply]
If you are able to give me permissions on the transferred repository (for some reason I dont inherit perms for some repos under the `getgauge` org) I can change the NPM publish key to be published by getgauge into the existing package and we can then test things for the next round, rather than publishing under your name/identity.

**Reply by zabil** [reply]
getgauge now has permissions to manage this package repo. I see the settings tab when on the getgauge user account

<img width="1324" alt="Screenshot 2024-06-07 at 10 31 32" src="https://github.com/getgauge/gauge/assets/54427/ffae5c58-fcb8-4205-9ac8-80bb2e71f9c3">
